### PR TITLE
Issue313 integration nc data reader

### PR DESCRIPTION
**Pull request by PMehrfeld**
See #313. 
Model for reader of external data implemented. NcDataReader represents an alternative to CombiTimeTable.
Since an external library is used, a check especially whether the example works or not is inevitable.
@mlauster, @marcusfuchs: I would suggest you for reviewer and assignee of this PR, respectively. Please suggest someone else to do the review, if you do not have time for this.

### CODE REVIEW

File: AixLib/Utilities/Sources/ncDataReader.mo [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        coordinateSystem(preserveAspectRatio=false)),
+        preferredView="info",
+    Documentation(revisions="<html>
+<ul>
+<li><i>Feburary 1, 2017</i> by Philipp Mehrfeld:<br>Revised documentation.</li>
+<li><i>December 14, 2016 </i> by Fabian Wuellhorst:<br>Implemented.</li>
+</ul>
+</html>", info="<html>
+<p><b><span style=\"color: #008000;\">Overview</span></b> </p>
+<p>The model <b>NcDataReader</b> reads data from netCDF files (based on HDF) to Modelica. You can read attributes of type integer or double/float/Real as well as one dimensional vectors. Therefore, the NcDataReader represents an alternative for the <a href=\"Modelica.Blocks.Sources.CombiTimeTable\">CombiTimeTable</a>.</p>
```
Comment (mlauster) [inline AixLib/Utilities/Sources/ncDataReader.mo]: Please shorten lines to max. 80 characters where applicable. HTML links and HTML commands are sometimes hard to break, so I would suggest to leave them as they are. But for the rest, it improves readability (you don't have to use the horizontal scroll bar) and simplifies life for GIT (which works line by line, so short lines help GIT to accurately locate changes).
The best way to do that?
1. install Atom editor
1. install language-modelica package in Atom
1. open your source folder in Atom
1. set line length in Atom to 80 (default value), you have a vertical grey line indicating that limit
1. break the lines manually
1. that workflow is also helpful when cleaning up code

File: AixLib/Utilities/Sources/ncDataReader.mo [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        fileName,
+        "",
+        attNameInt[i]);
+    end for;
+  else
+    for i in 1:(size(attNameInt))[1] loop
+      yAttInt_internal[i] = 0;
+    end for;
+  end if;
+     annotation (uses(NcDataReader2(version="2.4.0")),Dialog(group="AHU Modes"), choices(checkBox=true),Icon(coordinateSystem(preserveAspectRatio=false), graphics={
```
Comment (mlauster) [inline AixLib/Utilities/Sources/ncDataReader.mo]: Shorten line

File: AixLib/Utilities/Sources/ncDataReader.mo [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  parameter String fileName=Modelica.Utilities.Files.loadResource("modelica://NcDataReader2/Resources/examples/testfile.nc")
+    "File where external data is stored" annotation (Dialog(loadSelector(filter=
+           "netCDF files (*.nc)", caption="Open file")));
+  parameter Boolean use_varName = false "Get variables from fileName"
+       annotation(HideResult=True, Evaluate=true, choices(checkBox=true));
+  parameter Boolean use_attNameReal = false "Get real attributes from fileName"
+       annotation(HideResult=True, Evaluate=true, choices(checkBox=true));
+  parameter Boolean use_attNameInt = false "Get integer attributes from fileName"
+       annotation(HideResult=True, Evaluate=true, choices(checkBox=true));
+  parameter String varName[:]={"test1D"} "Array name in .nc file" annotation (Dialog(enable = use_varName));
```
Comment (mlauster) [inline AixLib/Utilities/Sources/ncDataReader.mo]: Might be pedantic, but for me it looks nicer if annotations and even descriptions get an own line. Again shortens the lines and cleans up the code.

File: AixLib/Utilities/Sources/ncDataReader.mo [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  parameter String attNameReal[:]={"foo"} "Name of attribute of type real in .nc file" annotation (Dialog(enable = use_attNameReal));
+  parameter String attNameInt[:]={"bar"} "Name of attribute of type integer in .nc file" annotation (Dialog(enable = use_attNameInt));
+  parameter Modelica.SIunits.Time offset = 0 "Time period prior current simulation time";
```
Comment (mlauster) [inline AixLib/Utilities/Sources/ncDataReader.mo]: Do the default values of all your parameters are really default values, applicable for typical cases? If so, keep them, if not, Modelica code is as well checkable without default values.

File: AixLib/Utilities/Examples/NcDataReader.mo [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    __Dymola_experimentFlags(
+      Advanced(GenerateVariableDependencies=false, OutputModelicaCode=false),
+      Evaluate=true,
+      OutputCPUtime=true,
+      OutputFlatModelica=false),
+    Icon(coordinateSystem(preserveAspectRatio=false)),
+    Diagram(coordinateSystem(preserveAspectRatio=false)),
+    Documentation(info="<html>
+<p><b><span style=\"color: #008000;\">Overview</span></b> </p>
+<p>This is an example for the model <a href=\"AixLib.Utilities.Sources.NcDataReader\">NcDataReader</a>. It simulates the air temperature of TRY 2010 region 12 (see attributes of the model <code>attNameInt</code>). The attribute <code>float_ex</code> is just a random value to show the use for attributes of type Real. It is also recommended to have a look into the source <code>*.cdl file</code> (modelica://AixLib/Resources/NcDataReader_ExampleData/Temp_Year.cdl).</p>
```
Comment (mlauster) [inline AixLib/Utilities/Examples/NcDataReader.mo]: Same about line length

**Comment by marcusfuchs** [comment]
As Github now allows for requesting reviews and assigning the PR separately, we should discuss how to handle this, as previously, reviewer and assignee were the same person in our workflow. What would my task be here after @mlauster approves the PR?

**Comment by PMehrfeld** [comment]
@marcusfuchs, @mlauster: Generally, in the future we can assign one person for both tasks. But in this case, the additional library `NcDataReader2`, which comes along with necessary external files that are included automatically by the data reader functions, is necessary. Therefore, it'll be really beneficious, if two persons test the new model. 
After I apply @mlauster's review comments, I will inform you @marcusfuchs so that you can just check the functionality with the example model. And probably also do a HTML check.

File: AixLib/Utilities/Sources/ncDataReader.mo (hunk 4 above)
Comment (PMehrfeld) [inline AixLib/Utilities/Sources/ncDataReader.mo]: done and also removed standard file name. offset = 0 should stay.

File: AixLib/Utilities/Sources/ncDataReader.mo (hunk 3 above)
Comment (PMehrfeld) [inline AixLib/Utilities/Sources/ncDataReader.mo]: done

File: AixLib/Utilities/Sources/ncDataReader.mo (hunk 2 above)
Comment (PMehrfeld) [inline AixLib/Utilities/Sources/ncDataReader.mo]: done

File: AixLib/Utilities/Sources/ncDataReader.mo (hunk 1 above)
Comment (PMehrfeld) [inline AixLib/Utilities/Sources/ncDataReader.mo]: Done where possible. Dymola also comes with vertical lines after 80 columns. But you are correct that I should use it...

File: AixLib/Utilities/Examples/NcDataReader.mo (hunk 5 above)
Comment (PMehrfeld) [inline AixLib/Utilities/Examples/NcDataReader.mo]: done

**Comment by PMehrfeld** [comment]
@mlauster: Are the line shortenings sufficient?
If so, @marcusfuchs (and if not done so far @mlauster) it would be good when you check whether you can simulate the example model or not.

**Comment by mlauster** [comment]
When I try to simulate `AixLib.Utilities.Examples.NcDataReader`, I get the following error message:

unbenanntes bild

Do I need to import some libraries or something similar? If so, it would be nice to have that information in the documentation of the NcDataReader model.

**Comment by PMehrfeld** [comment]
Yes, you need another library. Therefore, I referred to the library in the documentation (red font color, capital letters, and greater font size) and used the annotation key word `preferredView="info"`. I only did this in the NcDataReader.mo. Probably it might be better, to do this in the example model, too.
Here is the link [https://github.com/tbeu/netCDF-DataReader](https://github.com/tbeu/netCDF-DataReader)

If you have any ideas how to indicate the dependency to another library better, I will implement this, too. (I also use the annotation `uses(NcDataReader2(version="2.4.0"))`.)

**Comment by mlauster** [comment]
I saw that link, but what is missing is what I have to do to get it installed and running. I didn't check it in depth, what I found so far is:

configure
make
(make install)

Would it be possible to add that information (where do I have to put that library? Do I have to run some commands? Are there different commands for Windows, IOS and Linux? Do I need a restart? Do I have to add something to my environmental variables?)?

**Comment by PMehrfeld** [comment]
Thanks for this hint. I made it a bit clearer in the documentation that only the `netCDF-DataReader/NcDataReader2/package.mo` needs to be loaded in Modelica. No further installation is necessary.

**Comment by mlauster** [comment]
HTML looks fine.

**Comment by mlauster** [comment]
This closes #313

**Pull request merged**